### PR TITLE
checkin some test utils for tensorflow operation loader

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
checkin some test utils for tensorflow operation loader. It will compare the output from a single operation graph executed by BigDL and Tensorflow.

This PR also refine the test utility methods

## How was this patch tested?
unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Table.scala [hunk 1]
```diff
@@ -339,8 +339,8 @@ object T {
    * @param data
    * @return
    */
-  def array(data: Array[Any]): Table = {
-    new Table(data)
+  def array(data: Array[_]): Table = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Table.scala]: make the API easier to use

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/BigDLToTensorflow.scala [hunk 2]
```diff
@@ -139,8 +139,7 @@ object TemporalConvolutionToTF extends BigDLToTensorflow {
     require(inputs.length == 1, "SpatialConvolution only accept one input")
     val spatialConv = module.asInstanceOf[TemporalConvolution[_]]
 
-    val const1 = const(Tensor[Float](1).setValue(1, 1.0f),
-      spatialConv.getName() + "/dim1", byteOrder, true, DataType.DT_INT32)
+    val const1 = const(Tensor.scalar[Int](1), spatialConv.getName() + "/dim1", byteOrder)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/BigDLToTensorflow.scala]: refine the const method

**Comment by yiheng** [comment]
PR validation pass.

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowSpecHelper.scala [hunk 3]
```diff
@@ -57,4 +62,74 @@ class TensorflowSpecHelper extends FlatSpec with Matchers with BeforeAndAfter {
...
+    val bigdlOutput = runGraphBigDL(graphFile, nodeDefBuilder.getName)
+    val bigdlOutputTensor = if (bigdlOutput.isTensor) {
+      require(outputIndex == 0, s"invalid output index $outputIndex")
+      bigdlOutput.asInstanceOf[Tensor[_]]
+    } else {
+      bigdlOutput.toTable.apply[Tensor[_]](outputIndex + 1)
+    }
+    val tfOutput = runGraphTF(graphFile, nodeDefBuilder.getName + s":$outputIndex")
+    bigdlOutput.asInstanceOf[Tensor[NumericWildCard]]
+      .equals(tfOutput.asInstanceOf[Tensor[NumericWildCard]]) should be(true)
```
Comment (yangw1234) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowSpecHelper.scala]: Maybe we should use almostEqual and let the caller pass in the tolerance?

**Comment by yangw1234** [comment]
LGTM except one minor comment

**Pull request merged**